modules/numbers/shared/src/main/scala/io/circe/numbers/BiggerDecimal.scala

```diff
@@ -328,7 +328,7 @@ final object BiggerDecimal {
         i += 1
       }
 
-      if (state == FAILED) null else {
+      if (state == FAILED || state == AFTER_DOT || state == AFTER_E || state == AFTER_EXP_SIGN) null else {
```

Comment:
Not sure what style you prefer here, so open to suggestions. Could also do a match, possibly with `@switch` but that's probably overkill